src/Composer/Satis/Builder/Builder.php

```diff
@@ -47,4 +47,29 @@ public function __construct(OutputInterface $output, $outputDir, $config, $skipE
         $this->config = $config;
         $this->skipErrors = (bool) $skipErrors;
     }
+
+    /**
+     * Remove from $directory all files excluding $excludeFiles
+     *
+     * @param string          $directory    Directory to clean
+     * @param string|string[] $excludeFiles File list to exclude
+     *
+     * @return void
+     */
+    protected function clean($directory, $excludeFiles)
```

Comment:
I don't thinks this has any reason to be in the base class, as only one of the child classes actually needs it